### DISCUSSION TITLE
using glue::glue strings as targets

### DISCUSSION THREAD

**Discussion by jmbuhr** [ropensci/targets#821] [Help]
Hi there, the tidyverse `glue` package (https://glue.tidyverse.org/reference/glue.html) allows neat string interpolation with variable substitution such as

```r
x <- "hello"
glue::glue("{x} world")
```

however, due to the way theses strings are evaluated glue looks for theses variables in it's parent frame but is not able to find targets:

```r
list(
  tar_target(input, "hello"),
  tar_target(test, glue::glue("{input} world"))
)

#> object 'input' not found
```

One way around that is assigning the upstream target to a temporary variable:

```r
list(
  tar_target(input, "hello"),
  tar_target(test,{x <- input; glue::glue("{x} world")})
)
```

Is there a more elegant solution for this?

**Comment by wlandau** [comment]
In your first pipeline, the symbol `input` does not appear in the command of the target `test`, so `targets` does not know that `test` depends on `input`. Your workaround exposes the symbol and fixes the problem in that case. In theory it should be possible to write a function that automatically looks for `glue` strings and creates expressions with explicit symbols, but that's a lot of work. I do not know of other alternatives.

**Reply by nicki-dese** [reply]
Just wanted to add that glue actually makes it easy to work around this problem, because you can specify the variables you're calling within the function like so:

```{r}
list(
  tar_target(input, "hello"),
  tar_target(test, glue::glue("{x} world", x = input))
)
```
